### ISSUE TITLE
PostgreSQL sorting based on LC_COLLATE causes issues when not using `C`

### ISSUE THREAD

**Issue by fdietz** [richardboehme/lexorank#6]
First of all thanks a lot for this gem :) 

This is not really a bug report, but rather a summary of an issue I've stumbled across. I'm not an DB expert and therefore not 100% certain if all this is correct. But, maybe it helps others how run into the same issue.

After integrating your gem everything worked correctly in development/test environment on my local laptop. But, when deploying to Heroku on production it raised duplicate entry errors caused by the unique index on the "rank" column. This was indirectly caused by the wrong sorting order.

On Heroku PostgreSQL LC_COLLATE is set to `en_US.UTF-8`. But, on my laptop it is set to `C`.

When using `C` the sorting order is (I think) what we want, where uppercase letters come before lowercase letters. But, when using `en_US.UTF-8` the sorting order is different.

One can solve this at query time in sql `order by rank collate "C" asc` or in ruby `order(Arel.sql('rank collate "C" asc'))`. 

Alternatively, the column collate value can be changed with a migration:
```
ALTER TABLE items ALTER COLUMN rank TYPE varchar COLLATE "C";
```

**Comment by richardboehme** [comment]
Hey thanks for reaching out! 

As far as I can see from your response we would have two possibilities to improve the gem and fix this issue:

1. Add documentation on which collations are supported by the gem. This means everybody using this gem has to make sure they use a supported column collation. This gives the user the freedom on which collation they want to use, but may lead to unexpected errors if they do not take care about it.
2. Use `COLLATE` in our order query. I guess this would mean that we need to differentiate between different database adapters in the code because each DBMS uses different collation names. This would probably be the easiest way for users of the gem because they do not have to care about it at all.

Do you have a preferred way of fixing this? I think I personally prefer the second option. 

I definitely need to add multiple database adapters to the Github action script to verify they all work as expected.

**Comment by xanderificnl** [comment]
For people using Rails, this oughta be enough (in a migration) 

`t.text :rank, collation: 'C'`

Thanks for the gem. It helps!

**Comment by fdietz** [comment]
Sorry for my late reply!

1) I guess just adding this migration line (as proposed above) to the README for PostgreSQL users should be sufficient.
2) I'm still pretty new to the whole collation thingy. I've noticed that it doesn't work reliably when only adding the collation to the query. But, it works great if using a migration.

Thanks a lot!

**Comment by xanderificnl** [comment]
I'm guessing this problem isn't restricted to Postgres, has anyone tried it with MySQL?

**Comment by richardboehme** [comment]
Closed via #7.

Adding a new column type will be tracked in #8.
